test/e2e/common.go

```diff
@@ -0,0 +1,55 @@
+// copied from https://github.com/kubernetes-sigs/cluster-api/blob/v0.3.6/test/e2e/common.go
```

Comment:
I tried it before. but importing cluster-api many dependencies,and cluster-api test framework has unnecessary codes for secrets-store-csi-driver. e.g. we don't need to deploy cluster-api to test secrets-store-csi-driver.

Reply:
That's a fair concern. I would be interested to hear other's thoughts though, because personally I still have a slight preference to take a dependency - mainly for the benefit of getting upstream improvements and keeping work in this repo narrowly focussed on its main value-add proposition.

One mitigating factor with regards to dependencies - in this particular case it shouldn't inflate the size of the final published binary at all, because it's strictly the tests that are taking on additional dependencies.

In an ideal world, the framework in `cluster-api` would get split out into its own library at some point, but I appreciate that's not likely to happen any time soon, if at all.

Reply:
sounds good. Let me try to add `go.mod` inside `test/e2e` and follow latest cluster-api test framework. It would possible to remove files that I just copied.